File: worklist-ui/src/main/java/com/gcgenome/lims/client/domain/Serial.java
package com.gcgenome.lims.client.domain;

import jsinterop.annotations.JsPackage;
import jsinterop.annotations.JsType;
import lombok.experimental.Accessors;

@JsType(isNative=true, namespace= JsPackage.GLOBAL, name="Object")
@Accessors(fluent = true)
public final class Serial {
    public String id;
    public String worklist;
    public double index;
    public String serial;
    public String infix;
    public double idx;
}

File: domain/src/main/kotlin/com/gcgenome/lims/domain/Serial.kt
package com.gcgenome.lims.domain

import java.io.Serializable
import java.util.UUID

@JvmRecord
data class Serial (
    val id: UUID,
    val worklist: UUID,
    val index: Short,
    val serial: String,
    val infix: String,
    val idx: Short
): Serializable